x/ratelimit/keeper/rate_limit.go

```diff
@@ -19,13 +21,126 @@ func (k Keeper) GetChannelValue(ctx sdk.Context, denom string) sdk.Int {
 	return k.bankKeeper.GetSupply(ctx, denom).Amount
 }
 
-func CheckRateLimit(direction types.PacketDirection, packet exported.PacketI) error {
-	// TODO
-	return nil
+// QUESTION FOR REVIEWER: Where should these two functions live?
+
+// Parse the denom from the Send Packet that will be used by the rate limit module
+// The denom that the rate limiter will use for a SEND packet depends on whether
+//    it was a NATIVE token (e.g. ustrd, stuatom, etc.) or NON-NATIVE token (e.g. ibc/...)...
+//
+// We can identify if the token is native or not by parsing the trace denom from the packet
+// If the token is NATIVE, it will not have a prefix (e.g. ustrd),
+//    and if it is NON-NATIVE, it will have a prefix (e.g. transfer/channel-2/uosmo)
+//
+// For NATIVE denoms, return as is (e.g. ustrd)
+// For NON-NATIVE denoms, take the ibc hash (e.g. hash "transfer/channel-2/usoms" into "ibc/...")
+func ParseDenomFromSendPacket(packet transfertypes.FungibleTokenPacketData) (denom string) {
+	// Determine the denom by looking at the denom trace path
+	denomTrace := transfertypes.ParseDenomTrace(packet.Denom)
+
+	// Native assets will have an empty trace path and can be returned as is
+	if denomTrace.Path == "" {
+		denom = packet.Denom
+	} else {
+		// Non-native assets should be hashed
+		denom = denomTrace.IBCDenom()
+	}
+
+	return denom
+}
+
+// Parse the denom from the Send Packet that will be used by the rate limit module
```

Comment:
Parse the denom from the Send Packet that will be used by the rate limit module
```suggestion
// Parse the denom from the Recv Packet that will be used by the rate limit module
```

Reply:
done

Reply:
Great!